code/extensions/JSONTextExtension.php

```diff
@@ -158,5 +162,8 @@ public function updateCMSFields(\SilverStripe\Forms\FieldList $fields)
                 }
             }
         }
+        
+        $fields->addFieldsToTab("Root.Main", $hidden_fields_array);
+        
```

Comment:
Hi @alessandromarotta - thanks for your help here. Could you please provide some working code that shows how you're using this extension where versioning doesn't work and then does - using these changes? Thanks!

Reply:
Also, some changes have been pushed upstream to master, so you'll need to rebase your branch on top of those :-)